tree/ntuple/v7/src/RNTupleWriter.cxx

```diff
@@ -95,6 +96,10 @@ std::unique_ptr<ROOT::Experimental::RNTupleWriter>
 ROOT::Experimental::RNTupleWriter::Append(std::unique_ptr<RNTupleModel> model, std::string_view ntupleName, TFile &file,
                                           const RNTupleWriteOptions &options)
 {
+   const auto *fileClassName = file.IsA()->GetName();
+   if (strcmp(fileClassName, "TXMLFile") == 0)
+      throw RException(R__FAIL("RNTupleWriter doesn't support writing to an XML file"));
```

Comment:
```suggestion
   if (!file.IsBinary() == 0)
      throw RException(R__FAIL("RNTupleWriter only support writing to a binary file. Can not write into "+std::string(file.GetName()));
```
or if keeping the same logic
```suggestion
   static TClassRef XmlClass("TXMLFile");
   const auto *fileClassName = file.IsA()->GetName();
   if (XmlClass == file->IsA()) == 0)
      throw RException(R__FAIL("RNTupleWriter doesn't support writing to an XML file. Can not write into "+std::string(file.GetName()));
```

Reply:
Also a common text file is the JSON format.

Reply:
@pcanal but the TFile doesn't have special logic if you create it with a ".json" extension, does it?

Reply:
Right there is no  `TJsonFile`.